source/core/src/main/com/csse3200/game/screens/MiniGameMenuScreen.java

```diff
@@ -0,0 +1,193 @@
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+import com.badlogic.gdx.utils.viewport.ScreenViewport;
+import com.csse3200.game.GdxGame;
+
+public class MiniGameMenuScreen implements Screen {
+
+    private Stage stage;
+    private BitmapFont font;
+    private GdxGame game;
+    private Skin skin;
+
+    private Texture snakeTexture;
+    private Texture skyTexture;
+    private Texture waterTexture;
+
+    public MiniGameMenuScreen(GdxGame game) {
+        this.game = game;
+    }
+
+    @Override
+    public void show() {
+        stage = new Stage(new ScreenViewport());
+        Gdx.input.setInputProcessor(stage);
+
+        font = new BitmapFont();
+
+
+        font.getData().setScale(2);
+
+
+        skin = new Skin(Gdx.files.internal("uiskin.json"));
+
+
+        TextButtonStyle buttonStyle = new TextButtonStyle();
+        buttonStyle.font = font;
+        buttonStyle.up = skin.newDrawable("default-round", Color.BLUE);
+        buttonStyle.down = skin.newDrawable("default-round", Color.RED);
+        buttonStyle.disabled = skin.newDrawable("default-round", Color.GRAY);
+
+
+        snakeTexture = new Texture(Gdx.files.internal("images/minigames/Snake.png"));
+        skyTexture = new Texture(Gdx.files.internal("images/minigames/Flappy_bird.png"));
+        waterTexture = new Texture(Gdx.files.internal("images/minigames/Underwater_maze.png"));
+
+
+        Image snakeImage = new Image(snakeTexture);
+        Image skyImage = new Image(skyTexture);
+        Image waterImage = new Image(waterTexture);
+
+
+        TextButton snakeButton = new TextButton("Snake", buttonStyle);
+        TextButton skyButton = new TextButton("Flappy bird", buttonStyle);
+        TextButton waterButton = new TextButton("Underwater maze", buttonStyle);
+
+
+        Table table = new Table();
+        table.setFillParent(true);
+        table.center();
+
+
+        table.add(snakeImage).pad(10).row();
+        table.add(snakeButton).pad(10).row();
+        table.add(skyImage).pad(10).row();
+        table.add(skyButton).pad(10).row();
+        table.add(waterImage).pad(10).row();
+        table.add(waterButton).pad(10).row();
+
+
+        stage.addActor(table);
+
+
+        snakeImage.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                // Change button color to green when clicked
+                snakeImage.setColor(Color.GREEN);
+                // Switch to SnakeScreen
+                game.setScreen(new SnakeScreen(game));
+            }
+        });
+
+        skyImage.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                // Change button color to green when clicked
+                skyImage.setColor(Color.GREEN);
+                // Switch to SkyScreen
+              //  game.setScreen(new SkyScreen(game));
```

Comment:
remove commented code

Reply:
It is kept there for future implementation of other minigames.

Reply:
Shouldn't be in main; keep it in feature branch

Reply:
But we need the class to have our mini game even in the game in the first place. I can add a todo comment.

Reply:
Make a branch off this pull request branch that you'll use for additional minigame development that includes the commented code, and manually remove the commented code from this pull request. Work in progress code should only be kept in feature/development branches.